lib/mirage/impl/mirage_impl_kv.ml

```diff
@@ -11,8 +10,12 @@ let crunch dirname =
     | '0' .. '9' | 'a' .. 'z' | 'A' .. 'Z' -> true
     | _ -> false
   in
-  let modname = String.filter is_valid dirname in
-  let name = "Static_" ^ String.Ascii.lowercase modname in
+  let modname =
+    let b = Bytes.make (String.length dirname) '_' in
+    String.iteri (fun idx c -> if is_valid c then Bytes.set b idx c) dirname;
+    Bytes.unsafe_to_string b
+  in
```

Comment:
This differs but that's okay since we are only using this for making up names for the output. We can as well use `String.map`:
```suggestion
  let modname = String.map (fun c -> if is_valid c then c else '_') dirname in
```
I would furthermore suggest inlining `modname` as below it is being shadowed, and this is actually **not** the final module name. The `modname` as defined here *could* start with an underscore, and that would not be a legal OCaml module name.

Reply:
that's a great suggestion, done in 295fe8cf